From 1d89bc0004ec27fbc0c89f17861118c78d7eeab5 Mon Sep 17 00:00:00 2001
From: Dan Winship <danw@gnome.org>
Date: Fri, 4 Jan 2013 11:53:03 -0500
Subject: [PATCH] systemd: set KillMode=process

By default, when shutting NM down, systemd will kill everything in its
cgroup. But this can cause problems (eg, NM thinking that dhclient
crashed and then taking down an interface that it would otherwise have
left up). Fix this by setting KillMode=process, which tells systemd to
only kill NM itself, and let NM kill its children.

https://bugzilla.redhat.com/show_bug.cgi?id=876218
---
 data/NetworkManager.service.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/data/NetworkManager.service.in b/data/NetworkManager.service.in
index 9a5f125..16b53b6 100644
--- a/data/NetworkManager.service.in
+++ b/data/NetworkManager.service.in
@@ -12,6 +12,8 @@ ExecStart=@sbindir@/NetworkManager --no-daemon
 # with LOG_PERROR when run in foreground. But systemd redirects stderr to
 # syslog by default, which results in logging each message twice.
 StandardError=null
+# NM doesn't want systemd to kill its children for it
+KillMode=process
 
 [Install]
 WantedBy=multi-user.target
-- 
1.8.0.1

